### PR TITLE
Stop flushing cache on reads

### PR DESCRIPTION
**Pull request by szarsti**

### CODE REVIEW

File: apps/metric_vnode/src/metric_io.erl [hunk 1]
```diff
@@ -63,6 +65,9 @@ swrite(Pid, Bucket, Metric, Time, Value) ->
 read(Pid, Bucket, Metric, Time, Count, ReqID, Sender) ->
     gen_server:cast(Pid, {read, Bucket, Metric, Time, Count, ReqID, Sender}).
 
+read_rest(Pid, Bucket, Metric, Time, Count, Part, ReqID, Sender) ->
+    gen_server:cast(Pid, {read_rest, Bucket, Metric, Time, Count, Part, ReqID, Sender}).
```
Comment (elvisci) [inline apps/metric_vnode/src/metric_io.erl]: Line 69 is too long:     gen_server:cast(Pid, {read_rest, Bucket, Metric, Time, Count, Part, ReqID, Sender})..

File: apps/metric_vnode/src/metric_io.erl [hunk 2]
```diff
@@ -385,19 +390,29 @@ handle_cast({write, Bucket, Metric, Time, Value}, State) ->
...
+    {D, State1} = do_read(Bucket, Metric, Time, Count, State),
     riak_core_vnode:reply(Sender, {ok, ReqID, {P, N}, D}),
     {noreply, State1};
 
+handle_cast({read_rest, Bucket, Metric, Time, Count, Part, ReqID, Sender},
+            State = #state{node = N, partition = P}) ->
+    {Data, State1} =
+        case Part of
+            {Offset, PCount, Bin} when Offset =:= 0 ->
+                {{Res, D}, S} = do_read(Bucket, Metric, Time + PCount, Count - PCount, State),
```
Comment (elvisci) [inline apps/metric_vnode/src/metric_io.erl]: Line 402 is too long:                 {{Res, D}, S} = do_read(Bucket, Metric, Time + PCount, Count - PCount, State),.

File: apps/metric_vnode/src/metric_vnode.erl [hunk 3]
```diff
@@ -230,25 +230,26 @@ handle_command({get, ReqID, Bucket, Metric, {Time, Count}}, Sender,
...
-            ets:delete(T, {Bucket, Metric}),
-            metric_io:write(IO, Bucket, Metric, Start, Bin),
-            metric_io:read(IO, Bucket, Metric, Time, Count, ReqID, Sender),
+            PartStart = max(Time, Start),
+            PartCount = min(Time + Count, Start + Size) - PartStart,
+            SkipBytes = (PartStart - Start),
+            Bin = k6_bytea:get(Array, SkipBytes, (PartCount * ?DATA_SIZE)),
+            Offset = PartStart - Time,
+            Part = {Offset, PartCount, Bin},
+            metric_io:read_rest(IO, Bucket, Metric, Time, Count, Part, ReqID, Sender),
```
Comment (elvisci) [inline apps/metric_vnode/src/metric_vnode.erl]: Line 248 is too long:             metric_io:read_rest(IO, Bucket, Metric, Time, Count, Part, ReqID, Sender),.

File: apps/metric_vnode/src/metric_io.erl [hunk 4]
```diff
@@ -385,19 +390,29 @@ handle_cast({write, Bucket, Metric, Time, Value}, State) ->
...
+                {{Res, <<Bin/binary, D/binary>>}, S};
+            {Offset, PCount, Bin} when Offset + PCount =:= Count ->
+                {{Res, D}, S} = do_read(Bucket, Metric, Time, Count - PCount, State),
+                {{Res, <<D/binary, Bin/binary>>}, S};
+            {Offset, PCount, Bin} ->
+                {{Res, D}, S} = do_read(Bucket, Metric, Time, Count, State),
+                D1 = binary_part(D, 0, Offset * ?DATA_SIZE),
+                D2 = binary_part(D, Count * ?DATA_SIZE, (Offset + PCount - Count) * ?DATA_SIZE),
```
Comment (elvisci) [inline apps/metric_vnode/src/metric_io.erl]: Line 410 is too long:                 D2 = binary_part(D, Count * ?DATA_SIZE, (Offset + PCount - Count) * ?DATA_SIZE),.

File: apps/metric_vnode/src/metric_io.erl [hunk 5]
```diff
@@ -385,19 +390,29 @@ handle_cast({write, Bucket, Metric, Time, Value}, State) ->
...
 
+handle_cast({read_rest, Bucket, Metric, Time, Count, Part, ReqID, Sender},
+            State = #state{node = N, partition = P}) ->
+    {Data, State1} =
+        case Part of
+            {Offset, PCount, Bin} when Offset =:= 0 ->
+                {{Res, D}, S} = do_read(Bucket, Metric, Time + PCount, Count - PCount, State),
+                {{Res, <<Bin/binary, D/binary>>}, S};
+            {Offset, PCount, Bin} when Offset + PCount =:= Count ->
+                {{Res, D}, S} = do_read(Bucket, Metric, Time, Count - PCount, State),
```
Comment (elvisci) [inline apps/metric_vnode/src/metric_io.erl]: Line 405 is too long:                 {{Res, D}, S} = do_read(Bucket, Metric, Time, Count - PCount, State),.

**Comment by szarsti** [comment]
I made it aginst wrong branch. Will do it again aginst test.

**Pull request closed without merging**